doc/Caliptra.md

```diff
@@ -511,58 +511,47 @@ See [HW Section](#hardware-section) for a detailed discussion.
+6. SoC FMC continues the boot process, forming a boot firmware chain-of-trust: each firmware fetches/authenticates/measures/executes the next firmware needed to establish the device operating environment. Each firmware deposits the next firmware's measurements into Caliptra prior to execution.
+7. Caliptra firmware presents attestation APIs using the deposited measurements.
 
-In the Boot Media Integrated profile, the Caliptra Trusted Computing Base (TCB) for integrity of Core Root of Trust measurement is the Caliptra Core.  The verification of measurement includes:
+*Figure 3: Boot Media Integrated Boot Flow*
 
-	1. The SoC design ingests firmware through Caliptra
-	2. Caliptra IP, Caliptra ROM, and Caliptra Firmware
+![](./images/BMI_Boot_flow.jpeg)
 
 ### <a id="bmd"></a>Caliptra Core - Boot Media Dependent (BMD)
-
-To facilitate ease of integration and reliable measurement, the Caliptra RTM is the first uncore controller to be taken out of reset by the SoC ROM.  Once loaded, it provides callback signals for the remaining SoC subsystem to resume normal reset flow.
-
-All firmware that is loaded from an outside entity (and subsequently executed on the microprocessor) shall be considered untrusted; this firmware's measurements shall be reported to the Caliptra RTM before it is allowed to run within the SoC. Caliptra provides a HW API where the hash calculation, FW authentication and measurement capture is performed by Caliptra itself vs just reporting the measurements.
-
-In the BMD profile, the Caliptra trusted computing base (TCB) for integrity of Core Root of Trust measurement is the Caliptra Core and SoC ROM.  The verification of measurement mechanism includes:
-
-	1. Hardware executes SoC power-on reset logic.
-	2. SoC ROM executes, reads Caliptra firmware into Mailbox SRAM buffer.
-	3. SoC ROM signals Caliptra for its ROM to execute
-	4. SoC ROM pauses and waits for a signal (resume) from Caliptra
-	5. Caliptra ROM cryptographically authenticates its FW, measures and (if valid) executes its FW and then derives cryptographic identities
-	6. Caliptra signals back to SoC to resume reset.
-	7. SoC reads in its firmware, cryptographically authenticates its FW and provides measurements to Caliptra before executing.
+In the Boot Media Dependent Profile, Caliptra coordinates the start of the firmware chain-of-trust with the immutable component of the SoC ROM. Once the Caliptra ROM has completed initialization, it provides a "stash measurement" API and callback signals for the SoC ROM to proceed with the boot process. Caliptra ROM supports stashing of at most eight measurements prior to the boot of Caliptra firmware. The SoC then may choose to boot Caliptra firmware. Any security-sensitive code or configuration loaded by the SoC prior to Caliptra firmware boot must be stashed within Caliptra. If the SoC exceeds Caliptra ROM's measurement stash capacity, attestation must be disabled until next cold reset. The boot process is as follows:
```

Comment:
Is there going to be a limit on what the SOC can load?  This seems to imply that it can load, measure and run anything it wants up to 8 measurements and then load the Caliptra FW.  Would this scenario be a concern?

Reply:
Right. Prior to this change it was mandatory for Caliptra FW to be loaded before any other FW to facilitate measurement stashing of all FW. This change increases the count from 0 to 8. We had to draw a line somewhere because the ROM's event log output needs to be fixed size. 8 seems reasonable to me in terms of number of measurements that are in boot critical paths before wanting to boot up Caliptra FW.

Once Caliptra FW is loaded, unlimited measurements can be stashed but only to 24 TCIs per DPE iRoT design.

Reply:
The main concern that we have here is that this is different from the original intent of Caliptra being the first FW to execute.

Reply:
SOC can load, measure and run anything it wants up to 8 measurements before load the Caliptra FW ?
How are these 8 SoC's codes verified and authenticated ? Or do they need to be verified and authenticated ?
It would be SoC's responsibility to verify and authenticae these code ?

Reply:
SOC can load, measure and run anything it wants up to 8 measurements before load the Caliptra FW ?
How are these 8 SoC's codes verified and authenticated ? Or do they need to be verified and authenticated ?
It would be SoC's responsibility to verify and authenticae these code ?

Reply:
In the BMD Profile, Caliptra is not responsible for enforcing secure boot properties on firmware beyond its own

Reply:
> In the BMD Profile, Caliptra is not responsible for enforcing secure boot properties on firmware beyond its own

In this case, it looks like that Caliptra as the Root of Trust is not totally compatible with DICE ?
As DICE requires a layered boot sequence which the next layer always get verified on the previous one.

Reply:
No, TCG DICE requires measurement of layers, but not authentication of layers. https://trustedcomputinggroup.org/resource/dice-attestation-architecture/

Reply:
Marking this as resolved, no further comments from Derek or John.